### PR TITLE
[B] Close dropdown when clicking current community

### PR DESCRIPTION
**Pull request by 1aurend**

### CODE REVIEW

File: packages/frontend/components/atomic/BaseDropdown/BaseDropdown.tsx [hunk 1]
```diff
@@ -37,7 +38,8 @@ const BaseDropdown = React.forwardRef<HTMLButtonElement, Props>(
           role={undefined}
           aria-label={label}
         >
-          {children}
+          {typeof children === "function" &&
+            children({ onClick: closePopover })}
```
Comment (1aurend) [inline packages/frontend/components/atomic/BaseDropdown/BaseDropdown.tsx:42]: @jen-castiron Will you double check my types on this before I merge? Thanks!

File: packages/frontend/components/atomic/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -8,18 +8,24 @@ type BaseProps = React.ComponentProps<typeof BaseDropdown>;
...
-    return (
-      <Styled.Item key={i}>{item && React.cloneElement(item)}</Styled.Item>
-    );
-  });
+  const submenu = (onClick: React.MouseEventHandler | undefined) => {
+    return menuItems.map((item, i) => {
+      if (item === null) return null;
+      return (
+        <Styled.Item key={i}>
+          {item && React.cloneElement(item, { onClick })}
```
Comment (jen-castiron) [inline packages/frontend/components/atomic/Dropdown/Dropdown.tsx]: There was a small problem here where a child's click event handler was getting overwritten. So if you had a button in the menu, any on click event would be changed to the hide function. I moved the onClick to the dropdown wrapper 👍

File: packages/frontend/components/atomic/BaseDropdown/BaseDropdown.tsx (hunk 1 above)
Comment (jen-castiron) [reply packages/frontend/components/atomic/BaseDropdown/BaseDropdown.tsx]: This check works, but if the children was not a function, it wasn't rendered. I added a small fix. 👍

**Comment by jen-castiron** [comment]
I found a few small issues and added tiny fixes & comments. I'll merge this in once it finishes deploying 👍

**Comment by jen-castiron** [comment]
Side note - this might be a good issue to bring up at a frontendo confab. It comes up a few times, but fix is always a little tricky, especially for a11y requirements. Would be good to get this infront on Dana just to make sure we've covered our bases.

**Pull request merged**